### PR TITLE
Update fine-tune docs

### PR DESCRIPTION
**Pull request by stevhliu**
This is a first draft of the updated fine-tune tutorial. Main updates include:

- Preprocess a different dataset (users are probably very used to seeing `imdb` now) to include more diverse examples so users don't feel like they are doing the same thing over and over.
- Update the `Trainer` example to include evaluating accuracy before you fine-tune. It seems like a good practice to include evaluating metrics during training, so we should present it first. Let me know if this is not common or generally applicable, and we can revert it!
- Update Keras example to use `to_tf_dataset` to convert a dataset to a TensorFlow format (cc @Rocketknight1 please feel free to let me know if I'm missing anything!)

### CODE REVIEW

File: docs/source/training.mdx [hunk 1]
```diff
@@ -10,385 +10,351 @@ an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express o
...
-Then we will need to convert our datasets from before in standard `tf.data.Dataset`. Since we have fixed shapes,
-it can easily be done like this. First we remove the _"text"_ column from our datasets and set them in TensorFlow
-format:
+```py
+>>> from transformers import DataCollatorWithPadding
 
-```python
-tf_train_dataset = small_train_dataset.remove_columns(["text"]).with_format("tensorflow")
-tf_eval_dataset = small_eval_dataset.remove_columns(["text"]).with_format("tensorflow")
+>>> data_collator = DataCollatorWithPadding(tokenizer=tokenizer, return_tensors="tf")
```
Comment (sgugger) [inline docs/source/training.mdx:162]: We are not presenting this on the PyTorch side, and the dataset has been padded to the maximum length. So let's ignore the data collator in this tutorial and write another tutorial where we do the dynamic padding for both?

File: docs/source/training.mdx [hunk 2]
```diff
@@ -10,385 +10,351 @@ an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express o
...
 ```
 
-Then we convert everything in big tensors and use the `tf.data.Dataset.from_tensor_slices` method:
+<Tip>
 
-```python
-train_features = {x: tf_train_dataset[x] for x in tokenizer.model_input_names}
-train_tf_dataset = tf.data.Dataset.from_tensor_slices((train_features, tf_train_dataset["label"]))
-train_tf_dataset = train_tf_dataset.shuffle(len(tf_train_dataset)).batch(8)
+[`Trainer`] uses [`transformers.DataCollatorWithPadding`] by default so you don't have to explicitly specify it.
```
Comment (sgugger) [inline docs/source/training.mdx:167]:
```suggestion
[`Trainer`] uses [`DataCollatorWithPadding`] by default so you don't have to explicitly specify it.
```

That's only if the tokenizer is passed, which wasn't in the call before.

File: docs/source/training.mdx [hunk 3]
```diff
@@ -10,385 +10,351 @@ an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express o
...
+>>> from transformers import AutoModelForSequenceClassification
+model = AutoModelForSequenceClassification.from_pretrained("bert-base-cased", num_labels=5)
+```
+
+### Optimizer and learning rate scheduler
+
+Create an optimizer and learning rate scheduler to fine-tune the model with. Let's use the default optimizer used in [`Trainer`], [`AdamW`]:
+
+```py
+>>> from transformers import AdamW
```
Comment (sgugger) [inline docs/source/training.mdx:281]: I think we should move people out of our version of `AdamW` to use the one in torch.

**Review by sgugger — APPROVED** [review]
Thanks for revamping this tutorial. Since we're doing it, I think the data collator with padding should be removed from the TF section since it's not used in the other two. We can have a more advanced tutorial showing this.

File: docs/source/training.mdx (hunk 1 above)
Comment (stevhliu) [reply docs/source/training.mdx:162]: > So let's ignore the data collator in this tutorial

I agree, but we run into a little problem with `to_tf_dataset()` which requires the `collate_fn` parameter. So I think we need to introduce a data collator, but maybe we can use `DefaultDataCollator` instead of `DataCollatorWithPadding`? Sorry if this is what you meant, and I misunderstood! 😅

Comment (sgugger) [reply docs/source/training.mdx:162]: @Rocketknight1 Why does the function need a data collator when everything is rectangular?

Comment (Rocketknight1) [reply docs/source/training.mdx:162]: Hi, you're right that we could just use the default data collator for this! We do need a data collator of some kind, though, because they handle things like renaming label columns.

Comment (stevhliu) [reply docs/source/training.mdx]: Thanks, updated to use `DefaultDataCollator` instead!
